src/optimizer/rule_rewrite.cpp

```diff
@@ -0,0 +1,626 @@
+#include <memory>
+
+#include "catalog/column_catalog.h"
+#include "catalog/index_catalog.h"
+#include "catalog/table_catalog.h"
+#include "expression/group_marker_expression.h"
+#include "optimizer/operators.h"
+#include "optimizer/absexpr_expression.h"
+#include "optimizer/optimizer_metadata.h"
+#include "optimizer/properties.h"
+#include "optimizer/rule_rewrite.h"
+#include "optimizer/util.h"
+#include "type/value_factory.h"
+
+namespace peloton {
+namespace optimizer {
+
+// ===========================================================
+//
+// ComparatorElimination related functions
+//
+// ===========================================================
+ComparatorElimination::ComparatorElimination(RuleType rule, ExpressionType root) {
+  type_ = rule;
+
+  auto left = std::make_shared<Pattern>(ExpressionType::VALUE_CONSTANT);
+  auto right = std::make_shared<Pattern>(ExpressionType::VALUE_CONSTANT);
+  match_pattern = std::make_shared<Pattern>(root);
+  match_pattern->AddChild(left);
+  match_pattern->AddChild(right);
+}
+
+int ComparatorElimination::Promise(GroupExpression *group_expr,
+                                   OptimizeContext *context) const {
+  (void)group_expr;
+  (void)context;
+  return static_cast<int>(RulePriority::MEDIUM);
+}
+
+bool ComparatorElimination::Check(std::shared_ptr<AbstractNodeExpression> plan,
+                                  OptimizeContext *context) const {
+  (void)context;
+  (void)plan;
+  return true;
+}
+
+void ComparatorElimination::Transform(std::shared_ptr<AbstractNodeExpression> input,
+                                      std::vector<std::shared_ptr<AbstractNodeExpression>> &transformed,
+                                      UNUSED_ATTRIBUTE OptimizeContext *context) const {
+  (void)transformed;
+  (void)context;
```

Comment:
You already marked context as UNUSED_ATTRIBUTE on line 49. Take that macro out then.

Reply:
Resolved.